static/js/components/PaymentDisplay.js

```diff
@@ -0,0 +1,123 @@
+// @flow
+import React, { useState } from "react"
+import { path, sum } from "ramda"
+
+import { compose } from "redux"
+import { connect } from "react-redux"
+import { mutateAsync } from "redux-query"
+import queries from "../lib/queries"
+import { parsePrice } from "../util/util"
+
+import {
+  createForm,
+  formatPrice,
+  formatReadableDateFromStr
+} from "../util/util"
+
+import type { PaymentPayload } from "../flow/ecommerceTypes"
+import type { ApplicationDetail } from "../flow/applicationTypes"
+
+type Props = {
+  application: ?ApplicationDetail,
+  sendPayment: (payload: PaymentPayload) => Promise<void>
+}
+
+export function PaymentDisplay(props: Props) {
+  const [amount, setAmount] = useState(0)
+  const { application, sendPayment } = props
+  if (!application) {
+    return null
+  }
+
+  const run = application.bootcamp_run
+  const totalSpent = sum(
+    application.orders.map(order => order.total_price_paid)
+  )
+  const cost = application.price
+
+  return (
+    <div className="container">
```

Comment:
Any reason you're not using Formik here? We need validation messages for an empty field, a non-number value, and an invalid amount at the very least. We're already using Formik elsewhere

Reply:
No reason, I'll look into it